pkg/apis/build/v1beta1/source.go

```diff
@@ -59,6 +59,13 @@ type Git struct {
 	//
 	// +optional
 	CloneSecret *string `json:"cloneSecret,omitempty"`
+
+	// ShallowCloneDepth specifies the depth of the shallow clone.
+	// If not specified or set to 0, a full clone will be performed.
+	// Values greater than 0 will create a shallow clone with the specified depth.
+	//
+	// +optional
+	ShallowCloneDepth *int `json:"shallowCloneDepth,omitempty"`
```

Comment:
I would simplify the name of this field to "Depth". This matches the `--depth` flag in the `git` command line.

```suggestion
	// Depth specifies the git clone depth. If set to 0, cloning will include the full source history.
	// Values greater than 0 will create a shallow clone with the specified history depth.
	//
	// +optional
	Depth *int `json:"depth,omitempty"`
```